test/comparison/test_binning.py

```diff
@@ -282,33 +286,34 @@ def test_query_to_ref_pair_generator(self):
         bs_data.DB.create_in_mem()
 
         query_gbk = create_mock_gbk(0, bs_enums.SOURCE_TYPE.QUERY)
+        query_gbk.save_all()
         ref_gbks = [
             create_mock_gbk(i, bs_enums.SOURCE_TYPE.REFERENCE) for i in range(1, 4)
         ]
 
-        query_to_ref_pair_generator = QueryToRefRecordPairGenerator("mix", 1, "mix")
+        query_to_ref_pair_generator = QueryRecordPairGenerator("mix", 1, "mix")
         source_records = [query_gbk.region]
+
         for ref_gbk in ref_gbks:
+            ref_gbk.save_all()
             source_records.append(ref_gbk.region)
 
         query_to_ref_pair_generator.add_records(source_records)
 
         # expected edges    def test_ref_to_ref_pair_generator_first_iteration(self):
 
-        expected_pairs = []
-        for ref_gbk in ref_gbks:
-            expected_pair = (query_gbk.region, ref_gbk.region)
-            expected_pairs.append(expected_pair)
-
         # get all edges
         actual_pairs = list(query_to_ref_pair_generator.generate_pairs())
 
-        self.assertListEqual(expected_pairs, actual_pairs)
+        for ref_gbk in ref_gbks:
+            expected_pair = (query_gbk.region, ref_gbk.region)
+            self.assertIn(expected_pair, actual_pairs)
 
-    def test_ref_to_ref_pair_generator_first_iteration(self):
-        """Tests whether the RefTorefPairGenerator correctly generates a set of
+    def test_missing_query_pair_generator_first_iteration(self):
```

Comment:
looks like an integration test more than unit test.
can this be moved to test/integration/test_binning.py?